### PR TITLE
docs: improve locators guide

### LINKED ISSUE #17134: [docs] Extract selectors overview and best practices from selectors reference

### PR DESCRIPTION
**Pull request by dgozman**
Fixes #17134.

### CODE REVIEW

**Review by pavelfeldman — APPROVED** [review]

File: docs/src/locators.md [hunk 1]
```diff
@@ -188,121 +129,367 @@ await page.Locator("button").CountAsync();
...
+You can target the label with something like `text=Password` and perform the following actions on the password input:
+- `click` will click the label and automatically focus the input field;
+- `fill` will fill the input field;
+- `inputValue` will return the value of the input field;
+- `selectText` will select text in the input field;
+- `setInputFiles` will set files for the input field with `type=file`;
+- `selectOption` will select an option from the select box.
+
+```js
+// Fill the input by targeting the label.
```
Comment (debs-obrien) [inline docs/src/locators.md:273]: same here. I suggest to add text outside of code block so we only have to maintain the explanation once

File: docs/src/locators.md [hunk 2]
```diff
@@ -188,121 +129,367 @@ await page.Locator("button").CountAsync();
...
+<div data-test-id='product-card'>
+  <span>Product 2</span>
+  <button>Buy</button>
+</div>
+```
+
+You can now click a button for a particular product:
+
+```js
+// Find a product card that contains text "Product 2"
```
Comment (debs-obrien) [inline docs/src/locators.md:319]: i suggest adding this text outside of code block and just make it a sentence. Find a product....then....

File: docs/src/locators.md [hunk 3]
```diff
@@ -188,121 +129,367 @@ await page.Locator("button").CountAsync();
...
-var count = await rows.CountAsync()
-for (let i = 0; i < count; ++i)
-  Console.WriteLine(await rows.Nth(i).TextContentAsync());
+Playwright supports CSS and XPath selectors.
 
-// Pattern 3: resolve locator to elements on page and map them to their text content
-// Note: the code inside evaluateAll runs in page, you can call any DOM apis there
-var texts = await rows.EvaluateAllAsync("list => list.map(element => element.textContent)");
+```js
+// CSS
```
Comment (debs-obrien) [inline docs/src/locators.md:358]: Again same as above. put text outside of code blocks. easier to maintain

File: docs/src/locators.md [hunk 4]
```diff
@@ -188,121 +129,367 @@ await page.Locator("button").CountAsync();
...
+// XPath shorthand
+await page.Locator('//button').ClickAsync();
+```
+
+### Avoid locators tied to implementation
+
+XPath and CSS selectors can be tied to the DOM structure or implementation. These selectors can break when the DOM structure changes. Similarly, [`method: Locator.nth`], [`method: Locator.first`], and [`method: Locator.last`] are tied to implementation and the structure of the DOM, and will target the incorrect element if the DOM changes.
+
+```js
+// BAD PRACTICE: avoid long css or xpath chains
```
Comment (debs-obrien) [inline docs/src/locators.md:432]: same here. add text outside of code block

File: docs/src/locators.md [hunk 5]
```diff
@@ -367,6 +558,200 @@ await rowLocator
...
+  <span>Title</span>
+  #shadow-root
+    <span>Details</span>
+</x-badge>
+```
+
+You can locate in the same way as if the shadow root was not present at all:
+
+```js
+// Clicks <span>Details</span>
```
Comment (debs-obrien) [inline docs/src/locators.md:579]: add text explanations outside of code blocks so we only have to maintain once. can be a list here or a paragraph

File: docs/src/locators.md [hunk 6]
```diff
@@ -367,6 +558,200 @@ await rowLocator
...
+await Expect(page.Locator("x-badge")).ToContainTextAsync("Details");
+```
+
+
+## Lists
+
+You can also use locators to work with the element lists.
+
+```js
+// Locate elements, this locator points to a list.
```
Comment (debs-obrien) [inline docs/src/locators.md:639]: remove text explanations from code blocks

File: docs/src/locators.md [hunk 7]
```diff
@@ -367,6 +558,200 @@ await rowLocator
...
+
+### Picking specific element from a list
+
+If you have a list of identical elements, and the only way to distinguish between them is the order, you can choose a specific element from a list with [`method: Locator.first`], [`method: Locator.last`] or [`method: Locator.nth`].
+
+However, use these methods with caution. Often times, the page might change, and locator will point to a completely different element from the one you expected. Instead, try to come up with a unique locator that will pass the [strictness criteria](#strictness).
+
+
+```js
+// Click the third item in the list of products
```
Comment (debs-obrien) [inline docs/src/locators.md:731]: same here for explanations

**Pull request merged**